### PR TITLE
TDL-9856: ensure setup includes non-automatic balance transactions

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-9856: ensure setup includes non-automatic balance transactions
- Added test case to verify that we only sync payout transactions of payout which has "automatic" field as True, as we cannot get payout transaction of payout which are not marked as "automatic": True.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_automatic_payout_transactions.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+from utils import stripe_obj_to_dict, client, midnight
+from tap_tester import runner, connections
+from base import BaseTapTest
+
+def get_payouts():
```
Comment (dbshah1212) [inline tests/test_automatic_payout_transactions.py:5]: Add functional comment to describe the reason for adding this function.

File: tests/test_automatic_payout_transactions.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    """
+        Test case to verify that we only collect payout_transactions for payouts containing "automatic" field as "True"
+    """
+
+    @staticmethod
+    def name():
+        return "tt_stripe_automatic_payout_transactions"
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (dbshah1212) [inline tests/test_automatic_payout_transactions.py:33]: Add functional comment to describe the reason for adding this function.

File: tests/test_automatic_payout_transactions.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            # collect payout ids for all the payout transaction records
+            payout_transaction_payout_ids = set()
+            for record in records:
+                payout_transaction_payout_ids.add(record.get("payout_id"))
+
+            # verify that data exists for payouts with "automatic" field as "True" and "False"
+            self.assertTrue(self.payouts_with_automatic_true is not None)
+            self.assertTrue(self.payouts_with_automatic_false is not None)
+
+            # loop over all the payout ids from the payout transactions
```
Comment (dbshah1212) [inline tests/test_automatic_payout_transactions.py:84]: And append what we are verifying inside the loop.

File: tests/test_automatic_payout_transactions.py (hunk 1 above)
Comment (harshpatel4crest) [reply tests/test_automatic_payout_transactions.py:5]: Added comment.

File: tests/test_automatic_payout_transactions.py (hunk 2 above)
Comment (harshpatel4crest) [reply tests/test_automatic_payout_transactions.py:36]: Added comment.

File: tests/test_automatic_payout_transactions.py (hunk 3 above)
Comment (harshpatel4crest) [reply tests/test_automatic_payout_transactions.py]: Added comment.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

File: tests/test_automatic_payout_transactions.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    """
+        Return all the payouts (with pagination), to determine the automatic and non-automatic payouts
+    """
+    # list of all data to return
+    data = []
+    # api call of 1st page
+    stripe_obj = client["payouts"].list(limit=100, created={"gte": midnight})
+    dict_obj = stripe_obj_to_dict(stripe_obj)
+    # add data
+    data += dict_obj['data']
```
Comment (RushT007) [inline tests/test_automatic_payout_transactions.py:15]: Handle the exception where dict_obj can be empty if this test run without running all fields test. 

e.g. my test run failed throwing following exception.

```
======================================================================
ERROR: setUpClass (test_automatic_payout_transactions.AutomaticPayoutTransactionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/code/tap-stripe/tests/test_automatic_payout_transactions.py", line 44, in setUpClass
    payouts = get_payouts()
  File "/opt/code/tap-stripe/tests/test_automatic_payout_transactions.py", line 17, in get_payouts
    data += dict_obj['data']
KeyError: 'data'
```

**Review by RushT007 — CHANGES_REQUESTED** [review]
In doc string of tests/test_automatic_payout_transactions.py:test_run(), add the requirement that before running this test we need to run all fields test.

@KrisPersonal, @kspeer825: Alternative implementation we can try is, we can push add records implementation in. base class and whenever required add the records if not available already.

e.g. if `dict_obj = stripe_obj_to_dict(stripe_obj)` return empty data, we can call the add_records() function.

**Review by kspeer825 — APPROVED** [review]

File: tests/test_automatic_payout_transactions.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            Create list of automatic and non-automatic payouts used for assertion
+        """
+        # get all the payouts
+        payouts = get_payouts()
+
+        # create lists of payout ids containing automatic field as "True" and "False"
+        cls.payouts_with_automatic_true = []
+        cls.payouts_with_automatic_false = []
+        for record in payouts:
+            if "automatic" in record and record.get("automatic"):
```
[the remark this answers is not in the document]
Comment (harshpatel4crest) [reply tests/test_automatic_payout_transactions.py]: Updated.

File: tests/test_automatic_payout_transactions.py (hunk 4 above)
Comment (RushT007) [reply tests/test_automatic_payout_transactions.py:15]: When all fields test is not run below line returns the null

`dict_obj = stripe_obj_to_dict(stripe_obj)
   data += dict_obj['data']`

So can we please have try-catch around it and throw exception with appropriate message?

**Review by RushT007 — CHANGES_REQUESTED** [review]

**Comment by harshpatel4crest** [comment]
@RushT007 Added try-catch for empty payouts records and raise error with a message.

**Review by RushT007 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**